crates/trie/db/src/witness.rs

```diff
@@ -8,39 +8,39 @@ use reth_trie::{
 };
 
 /// Extends [`TrieWitness`] with operations specific for working with a database transaction.
-pub trait DatabaseTrieWitness<'a, TX> {
+pub trait DatabaseTrieWitness {
     /// Create a new [`TrieWitness`] from database transaction.
-    fn from_tx(tx: &'a TX) -> Self;
+    fn from_tx(&self) -> Self;
 
     /// Generates trie witness for target state based on [`TrieInput`].
     fn overlay_witness(
-        tx: &'a TX,
+        &self,
         input: TrieInput,
         target: HashedPostState,
     ) -> Result<B256Map<Bytes>, TrieWitnessError>;
 }
 
-impl<'a, TX: DbTx> DatabaseTrieWitness<'a, TX>
+impl<'a, TX: DbTx> DatabaseTrieWitness
     for TrieWitness<DatabaseTrieCursorFactory<'a, TX>, DatabaseHashedCursorFactory<'a, TX>>
 {
-    fn from_tx(tx: &'a TX) -> Self {
-        Self::new(DatabaseTrieCursorFactory::new(tx), DatabaseHashedCursorFactory::new(tx))
+    fn from_tx(&self) -> Self {
+        Self::new(self.trie_cursor_factory.clone(), self.hashed_cursor_factory.clone())
     }
 
     fn overlay_witness(
-        tx: &'a TX,
+        &self,
         input: TrieInput,
         target: HashedPostState,
     ) -> Result<B256Map<Bytes>, TrieWitnessError> {
         let nodes_sorted = input.nodes.into_sorted();
         let state_sorted = input.state.into_sorted();
-        Self::from_tx(tx)
+        Self::from_tx(&self)
             .with_trie_cursor_factory(InMemoryTrieCursorFactory::new(
-                DatabaseTrieCursorFactory::new(tx),
+                self.trie_cursor_factory.clone(),
                 &nodes_sorted,
             ))
             .with_hashed_cursor_factory(HashedPostStateCursorFactory::new(
-                DatabaseHashedCursorFactory::new(tx),
+                self.hashed_cursor_factory.clone(),
```

Comment:
I think this actually motivates changing the `&self`, to be `self` in the trait, that way we could do something like:
```
self.
// with_trie_cursor_factory call
// with_hashed_cursor_factory call
// ...
.compute(target)
```